quint/src/quintParserFrontend.ts

```diff
@@ -114,97 +114,91 @@ export function parsePhase1(text: string, sourceLocation: string): ParseResult<P
  * Phase 2 of the Quint parser. Read the IR and check that all names are defined.
  * Note that the IR may be ill-typed.
  */
-export function parsePhase2(phase1Data: ParserPhase1):
-  ParseResult<ParserPhase2> {
-  const quintModule: QuintModule = phase1Data.module
+export function parsePhase2(phase1Data: ParserPhase1): ParseResult<ParserPhase2> {
   const sourceMap: Map<bigint, Loc> = phase1Data.sourceMap
-  const scopeTree = treeFromModule(quintModule)
-  const moduleDefinitions = collectDefinitions(quintModule)
-  const importResolvingResult = resolveImports(quintModule, moduleDefinitions)
-  let definitions: LookupTableByModule
-  const errorMessages: ErrorMessage[] = []
 
-  if (importResolvingResult.kind === 'ok') {
-    definitions = importResolvingResult.definitions
-  } else {
-    definitions = moduleDefinitions
-    importResolvingResult.errors.forEach(error => {
-      const sourceLoc = sourceMap.get(error.reference)
-      if (!sourceLoc) {
-        console.error(`No source location found for ${error.reference}. Please report a bug.`)
-      }
-      const loc = sourceLoc ?? unknownLoc
-      if (error.defName) {
-        const e =
-          `Failed to import definition ${error.defName} from module ${error.moduleName}`
-        errorMessages.push({
-          explanation: e,
-          locs: [loc],
-        })
-      } else {
-        errorMessages.push({
-          explanation: `Failed to import module ${error.moduleName}`,
-          locs: [loc],
-        })
-      }
-    })
-  }
+  return phase1Data.modules.reduce((result: ParseResult<ParserPhase2>, module) => {
+    const scopeTree = treeFromModule(module)
+    let definitions = collectDefinitions(module)
+    result.map(r => definitions = new Map([...r.table.entries(), ...definitions.entries()]))
```

Comment:
if i am reading correctly, if `result` is a `left` here we just ignore the failure and continue working with the `module`? That's surprising to me. Also doesn't the `ParseResult<ParserPhase2>` already have the phase 1 modules? So what is the new `module` argument here for?